core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -333,16 +333,16 @@ class DAGScheduler(
       // (so we don't unnecessarily re-compute that data).
       val serLocs = mapOutputTracker.getSerializedMapOutputStatuses(shuffleDep.shuffleId)
       val locs = MapOutputTracker.deserializeMapStatuses(serLocs)
-      (0 until locs.length).foreach { i =>
-        if (locs(i) ne null) {
-          // locs(i) will be null if missing
-          stage.addOutputLoc(i, locs(i))
+      locs.zipWithIndex
```

Comment:
I'd be wary of making this type of change since there are cases where we've purposely removed Scala-style transformations in hot scheduler loops.